aten/src/ATen/ParallelNative.cpp

```diff
@@ -178,10 +178,19 @@ void set_num_threads(int nthreads) {
 #ifndef C10_MOBILE
   TORCH_CHECK(nthreads > 0, "Expected positive number of threads");
   int no_value = NOT_SET;
-  TORCH_CHECK(num_intraop_threads.compare_exchange_strong(no_value, nthreads),
-      "Error: cannot set number of intraop threads "
-      "after parallel work has started or after set_num_threads call "
-      "when using native parallel backend");
+  if (!num_intraop_threads.compare_exchange_strong(no_value, nthreads)) {
```

Comment:
So now calling, for example, set_num_threads(2) twice doesn't raise exception but set_num_threads(2) and then again set_num_threads(4) still raises exception with native thread pool. Correct?

Reply:
yes, this is correct. The main reason for this is that we actually do have places that call set_num_threads(x) multiple times (where x is usually 1), and as of now, we may assume this to be a supported behavior. Changing number of threads though is not supported in native backend